components/browser/state/src/main/java/mozilla/components/browser/state/reducer/WebExtensionReducer.kt

```diff
@@ -46,6 +46,18 @@ internal object WebExtensionReducer {
                     it.copy(browserAction = action.browserAction)
                 }
             }
+            is WebExtensionAction.UpdateWebExtension -> {
+                val updatedExtension = action.updatedExtension
+                state.updateWebExtensionState(updatedExtension.id) { existingExtension ->
+                    existingExtension.copy(
+                        url = updatedExtension.url ?: existingExtension.browserActionPopupSession,
+                        browserAction = updatedExtension.browserAction
+                            ?: existingExtension.browserAction,
+                        browserActionPopupSession = updatedExtension.browserActionPopupSession
+                            ?: existingExtension.browserActionPopupSession
+                    )
```

Comment:
I'm not sure, but I have hunch that could be missing something here

Reply:
Yeah, there will be many more fields in the `WebExtension` object we get from GV, and we will get a new instance of the native `WebExtension` as well. So, I think it will be better to remove the old global `WebExtensionState` and add the new one based on the new object.

Reply:
👍